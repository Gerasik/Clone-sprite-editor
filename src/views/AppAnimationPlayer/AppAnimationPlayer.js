export default class AppAnimationPlayer {
  constructor(className, parentBlock, sizeCanvas) {
    this.className = className;
    this.parentBlock = parentBlock;
    this.sizeCanvas = sizeCanvas;
  }

  render() {
    const animationPlayer = document.createElement('section');
    animationPlayer.classList.add(this.className);
    animationPlayer.innerHTML = `<canvas class="${this.className}__player" width="${this.sizeCanvas.width}" height="${this.sizeCanvas.height}"></canvas>
                                <span class="${this.className}__full-size">&#xf066</span>
                                <span class="${this.className}__number">FPS: 12</span>
                                <input class="${this.className}__frequency" type="range" min="1" max="24" value="12" step="1">`;
    this.parentBlock.appendChild(animationPlayer);
    return animationPlayer;
  }

  updateFPS(e) { // animation player FPS counter input
    const frequency = e.target.value;
    const prevElem = e.target.previousElementSibling;
    prevElem.innerText = `FPS: ${frequency}`;
    this.frequency = frequency;
  }
}
