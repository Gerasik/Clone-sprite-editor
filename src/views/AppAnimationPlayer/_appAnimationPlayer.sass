.animation-player
  background-color: white
  padding: 10px
  align-self: flex-start
  width: 225px
  position: relative
  &:-webkit-full-screen
    .animation-player
      &__player
        height: 100%
      &__number,
      &__frequency
        display: none
  &__player
    border: 1px solid black
    margin: 0 auto
    width: 100%
    image-rendering: pixelated
  &__full-size
    font-family: 'Awesome'
    position: absolute
    top: 10px
    right: 10px
    font-size: 24px
    padding: 5px
    display: none
    background-color: #fff
    cursor: pointer
  &:hover
    .animation-player__full-size
      display: inline-block
