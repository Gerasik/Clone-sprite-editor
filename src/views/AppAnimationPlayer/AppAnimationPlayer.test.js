import AppAnimationPlayer from './AppAnimationPlayer';

const context = {
  className: 'player',
  parentBlock: document.body,
  sizeCanvas: { width: 20, height: 20 },
};

const newObj = new AppAnimationPlayer(context.className, context.parentBlock, context.sizeCanvas);
describe('new AppAnimationPlayer', () => {
  it('Should be an instance of Object', () => {
    expect(newObj).toBeInstanceOf(Object);
  });
  it('Should be contain right parameters', () => {
    expect(newObj.className).toBe('player');
  });
});

describe('AppAnimationPlayer.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppAnimationPlayer.prototype.render).toBeInstanceOf(Function);
  });

  const ans = newObj.render();
  it('Should be render container', () => {
    expect(document.body.innerHTML).toMatchSnapshot();
  });
  it('Should be render 4 child', () => {
    expect(ans.children.length).toBe(4);
  });
});

describe('AppAnimationPlayer.prototype.updateFPS', () => {
  it('Should be an instance of Function', () => {
    expect(AppAnimationPlayer.prototype.updateFPS).toBeInstanceOf(Function);
  });
  const cont = {
    frequency: null,
  };
  const event = {
    target: { value: 5, previousElementSibling: document.body },
  };
  AppAnimationPlayer.prototype.updateFPS.bind(cont, event)();
  it('Should be equival event.value', () => {
    expect(cont.frequency).toBe(event.target.value);
  });
});
