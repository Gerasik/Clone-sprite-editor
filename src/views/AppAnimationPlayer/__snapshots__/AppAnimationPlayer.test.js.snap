// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAnimationPlayer.prototype.render Should be render container 1`] = `
"<section class=\\"player\\"><canvas class=\\"player__player\\" width=\\"20\\" height=\\"20\\"></canvas>
                                <span class=\\"player__full-size\\">ÔÅ¶</span>
                                <span class=\\"player__number\\">FPS: 12</span>
                                <input class=\\"player__frequency\\" type=\\"range\\" min=\\"1\\" max=\\"24\\" value=\\"12\\" step=\\"1\\"></section>"
`;
